test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1675 @@
+                 log_path=None, sudo=False):
+        """
+        Initialize a PBSSnapUtils object with the arguments specified
+
+        :param out_dir: path to the directory where snapshot will be created
+        :type out_dir: str
+        :param diag_path: Path to the diag directory to create snapshot from
+        :type diag_path: str or None
+        :param server_host: Name of the host where PBS Server is running
+        :type server_host: str or None
+        :param acct_logs: number of accounting logs to capture
+        :type acct_logs: int or None
+        :param daemon_logs: number of daemon logs to capture
+        :type daemon_logs: int or None
+        :param additional_hosts: additional hosts to capture information from
+        :type additional_hosts: str or None
+        :param map_file: Path to map file for anonyization map
+        :type map_file str or None
+        :param anonymize: anonymize data?
+        :type anonymize: bool
+        :param create_tar: Create a tarball of the output snapshot?
+        :type create_tar: bool or None
+        :param log_path: Path to pbs_snapshot's log file
+        :type log_path: str or None
+        :param sudo: Capture information with sudo?
+        :type sudo: bool
+        """
+        self.logger = logging.getLogger(__name__)
+        self.du = DshUtils()
+        self.snap_path = {}
+        self.server_info = {}
+        self.job_info = {}
+        self.node_info = {}
+        self.comm_info = {}
+        self.hook_info = {}
+        self.sched_info = {}
+        self.resv_info = {}
+        self.pbs_info = {}
+        self.sys_info = {}
+        self.core_info = {}
+        self.datastore_info = {}
+        self.anon_obj = None
+        self.all_hosts = []
+        self.server = None
+        self.scheduler = None
+        self.log_utils = PBSLogUtils()
+        self.outtar_path = None
+        self.outtar_fd = None
+        self.create_tar = create_tar
+        self.snapshot_name = None
+        self.sudo = sudo
+        self.log_path = log_path
+        self.ctime_file = "ctime"
+        if self.log_path is not None:
+            self.log_filename = os.path.basename(self.log_path)
+        else:
+            self.log_filename = None
+
+        # finalize() is called by the context's __exit__() automatically
+        # however, finalize() is non-reenterant, so set a flag to keep
+        # track of whether it has been called or not.
+        self.finalized = False
+
+        # Parse the input arguments
+        self.diag = diag_path
+        timestamp_str = time.strftime("%Y%m%d_%H_%M_%S")
+        self.snapshot_name = "snapshot_" + timestamp_str
+        # Make sure that the target directory exists
+        dir_path = os.path.abspath(out_dir)
+        if not os.path.isdir(dir_path):
+            self.logger.error("Target directory either doesn't exist" +
+                              "or not accessible. Quitting.")
+            raise
```

Comment:
No, I wanted to raise an exception so that the program quits. I've changed it to now raise a ValueError instead as just doing a 'raise' is incorrect in this context. Thanks for catching this.